### PR TITLE
Fix insert or query by precise value error when there is no db ShardingStrategy and using unique table names

### LINKED ISSUE #10746: Insert error with IntervalShardingAlgorithm when no db route configs in multi datatources and datanode name is unique
### Which version of ShardingSphere did you use?
master branch

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?
ShardingSphere-Proxy & governance

### Expected behavior
table columns:
-  id(bigint)
-  create_at(datetime)

This is my config rules:
``` 
- !SHARDING
  tables:
    fixed_test:
      actualDataNodes: ds_0.fixed_test_${0..1},ds_1.fixed_test_${2..3}
      tableStrategy:
        standard:
          shardingAlgorithmName: fixed_test_hash_mod
          shardingColumn: id
      keyGenerateStrategy:
        column: id
        keyGeneratorName: snowflake
    range_test:
      actualDataNodes: ds_0.range_test_${0..4},ds_1.range_test_${5..9}
      tableStrategy:
        standard:
          shardingAlgorithmName: range_test_volume_range
          shardingColumn: level
      keyGenerateStrategy:
        column: id
        keyGeneratorName: snowflake
    time_test:
      actualDataNodes: ds_0.time_test_2021062,ds_0.time_test_2021063,ds_1.time_test_2021064,ds_1.time_test_2021071
      tableStrategy:
        standard: 
          shardingAlgorithmName: time_test_interval
          shardingColumn: create_at
      keyGenerateStrategy:
        column: id
        keyGeneratorName: snowflake
  shardingAlgorithms:
    fixed_test_hash_mod:
      type: HASH_MOD
      props:
        sharding-count: 4
    range_test_volume_range:
      type: VOLUME_RANGE
      props:
        range-lower: 1
        range-upper: 1000
        sharding-volume: 100
    time_test_interval:
      type: INTERVAL
      props:
        datetime-lower: '2021-06-13 00:00:01'
        datetime-interval-unit: weeks
        datetime-interval-amount: 1
        datetime-pattern: yyyy-MM-dd HH:mm:ss
        sharding-suffix-pattern: yyyyMMW
  keyGenerators:
    snowflake:
      type: SNOWFLAKE
    uuid:
      type: uuid

```  

sql:
``` 
INSERT INTO `time_test` (`create_at`) VALUES ('2021-06-13 03:01:01');
```
expect insert success

### Actual behavior
<e>查询：insert into `time_test` (`create_at`) values ('2021-06-13 03:01:01')

错误代码： 1999
Unknown exception: [failed to shard value PreciseShardingValue(logicTableName=time_test, columnName=create_at, value=2021-06-13 03:01:01), and availableTables [time_test_2021064, time_test_2021071]]

### Reason analyze (If you can)
image
why:
because there is no db strategy,so both ds_0 and ds_1 will enter this code span and make this error finally.
ps:
doSharding with PreciseShardingValue should only return one datanodes

### Example codes for reproduce this issue (such as a github link).

IntervalShardingAlgorithm do not throw any exception ,and check preciseSharding result at parent invoke span.

If you think this is a problem,I`d like to resolve it

### PR DESCRIPTION
**Pull request by jianliu**
Fixes #10746.

Changes proposed in this pull request:
- IntervalShardingAlgorithm`s method doSharding(...) do not throw exception directly,if no matched target just return null
- Optimized code of  IntervalShardingAlgorithm and AutoIntervalShardingAlgorithm ,reduce the times of dataTime parser

### CODE REVIEW

**Review by strongduanmu — COMMENTED** [review]
@jianliu Can you add some unit tests?

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/datetime/AutoIntervalShardingAlgorithm.java [hunk 1]
```diff
@@ -84,8 +84,9 @@ private long getShardingSeconds() {
     
     @Override
     public String doSharding(final Collection<String> availableTargetNames, final PreciseShardingValue<Comparable<?>> shardingValue) {
+        String shardingValSuffix = String.valueOf(doSharding(parseDate(shardingValue.getValue())));
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/datetime/AutoIntervalShardingAlgorithm.java:87]: @jianliu tableNameSuffix may be better.

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/datetime/IntervalShardingAlgorithm.java [hunk 2]
```diff
@@ -121,9 +120,10 @@ private ChronoUnit getStepUnit(final String stepUnit) {
     
     @Override
     public String doSharding(final Collection<String> availableTargetNames, final PreciseShardingValue<Comparable<?>> shardingValue) {
+        String shardingValSuffix = parseDateTime(shardingValue.getValue().toString()).format(tableSuffixPattern);
```
Comment (strongduanmu) [inline shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/datetime/IntervalShardingAlgorithm.java:123]: @jianliu tableNameSuffix may be better.

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/datetime/AutoIntervalShardingAlgorithm.java (hunk 1 above)
Comment (jianliu) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/datetime/AutoIntervalShardingAlgorithm.java:87]: ok,I`ll change it.

File: shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/datetime/IntervalShardingAlgorithm.java (hunk 2 above)
Comment (jianliu) [reply shardingsphere-features/shardingsphere-sharding/shardingsphere-sharding-core/src/main/java/org/apache/shardingsphere/sharding/algorithm/sharding/datetime/IntervalShardingAlgorithm.java:123]: ok,I`ll change it.

**Review by strongduanmu — APPROVED** [review]

**Comment by strongduanmu** [comment]
@jianliu Great job, merged. 👍

**Pull request merged**